components/eamxx/src/control/atmosphere_driver.cpp

```diff
@@ -1283,6 +1288,56 @@ void AtmosphereDriver::set_initial_conditions ()
     }
   }
 
+  // Compute IC perturbations of GLL fields (if requested)
+  using vos = std::vector<std::string>;
+  const auto perturbed_fields = ic_pl.get<vos>("perturbed_fields", {});
+  const auto num_perturb_fields = perturbed_fields.size();
+  if (num_perturb_fields > 0) {
+    m_atm_logger->info("    [EAMxx] Adding random perturbation to ICs ...");
+
+    EKAT_REQUIRE_MSG(m_field_mgrs.count("Physics GLL") > 0,
+                     "Error! Random perturbation can only be applied to fields on "
+                     "the GLL grid, but no Physics GLL FieldManager was defined.\n");
+    const auto& fm = m_field_mgrs.at("Physics GLL");
+
+    // Setup RNG. If no random seed is requested in
+    // initial_conditions params, generate one using rand().
+    const auto seed = ic_pl.get<int>("perturbation_random_seed", rand());
+    std::mt19937_64 engine(seed);
+    m_atm_logger->info("      For IC perturbation, random seed: "+std::to_string(seed));
+
+    // Get perterbation limit. Defines a range [-perturbation_limit, perturbation_limit]
```

Comment:
Typo "perterbation"

Reply:
More importantly - awesome to print the random seed to the log file. Could we add the option for a user to specify the random seed for a given run? Then we could repeat a simulation if it did something weird or we didn't get the output we needed or something.

Reply:
The user can request put `perturbation_random_seed:` in the input.yaml. But I think the field values will only line up if the user also requests the same fields to be perturbed.

Reply:
>The user can request put perturbation_random_seed: in the input.yaml. But I think the field values will only line up if the user also requests the same fields to be perturbed.

That's fine. Someone who asks for different fields to be perturbed and expects to get the same answer has bigger problems.